### PR TITLE
Run helix matrix tests on AlmaLinux 8

### PR DESCRIPTION
**Pull request by sbomer**
This uses the new AlmaLinux 8 images added in https://github.com/dotnet/dotnet-buildtools-prereqs-docker/pull/860 for testing.

helix-matrix run: https://dev.azure.com/dnceng-public/public/_build/results?buildId=256645&view=results

### CODE REVIEW

File: src/Testing/src/xunit/HelixConstants.cs [hunk 1]
```diff
@@ -8,5 +8,5 @@ public static class HelixConstants
     public const string Windows10Arm64 = "Windows.10.Arm64v8.Open;";
     public const string DebianAmd64 = "Debian.11.Amd64.Open;";
     public const string DebianArm64 = "Debian.11.Arm64.Open;";
-    public const string RedhatAmd64 = "Redhat.7.Amd64.Open;";
+    public const string AlmaLinuxAmd64 = "(AlmaLinux.8.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:almalinux-8-helix-amd64;";
```
Comment (amcasey) [inline src/Testing/src/xunit/HelixConstants.cs:11]: I have no idea whether this is correct, but it _is_ notably different from the others.  In particular, why would it mention both alma and ubuntu?

File: src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs [hunk 2]
```diff
@@ -1047,7 +1047,7 @@ void ConfigureListenOptions(ListenOptions listenOptions)
     [ConditionalFact]
     [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "Missing platform support.")]
     [TlsAlpnSupported]
-    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/33566#issuecomment-892031659", Queues = HelixConstants.RedhatAmd64)] // Outdated OpenSSL client
+    [SkipOnHelix("https://github.com/dotnet/aspnetcore/issues/33566#issuecomment-892031659", Queues = HelixConstants.AlmaLinuxAmd64)] // Outdated OpenSSL client
```
Comment (amcasey) [inline src/Servers/Kestrel/test/InMemory.FunctionalTests/HttpsConnectionMiddlewareTests.cs:1050]: Not for this PR, but does alma also have an outdated OpenSSL client?

File: src/Testing/src/xunit/HelixConstants.cs (hunk 1 above)
Comment (amcasey) [reply src/Testing/src/xunit/HelixConstants.cs:11]: Fancy

**Review by amcasey — APPROVED** [review]
Seems good to me, but it might be good to hear from someone with more context (cough @wtgodbe).

File: eng/targets/Helix.Common.props [hunk 3]
```diff
@@ -37,7 +37,7 @@
       <!-- aspnetcore-quarantined-tests (quarantined-tests.yml) and RunHelix.ps1 -RunQuarantinedTests -->
       <ItemGroup>
         <!-- Linux -->
-        <HelixAvailableTargetQueue Include="Redhat.7.Amd64.Open" Platform="Linux" />
+        <HelixAvailableTargetQueue Include="(AlmaLinux.8.Amd64.Open)Ubuntu.1804.Amd64.Open@mcr.microsoft.com/dotnet-buildtools/prereqs:almalinux-8-helix-amd64" Platform="Linux" />
```
Comment (BrennanConroy) [inline eng/targets/Helix.Common.props:40]: Use a variable like https://github.com/dotnet/aspnetcore/pull/47970/files#diff-fecf6f98b5c2a5cc5c66f045a93d30421b7fedcbdd4cacefc2605cd61a9b89afL8? Then you don't need to copy this string across files

**Comment by amcasey** [comment]
All good @BrennanConroy?

**Review by BrennanConroy — APPROVED** [review]

**Review by wtgodbe — APPROVED** [review]

**Pull request merged**